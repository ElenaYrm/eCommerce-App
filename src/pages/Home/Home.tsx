import { ReactElement, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import classnames from 'classnames';
import { selectIsNewUser } from '../../store/auth/selectors';
import { useAppDispatch } from '../../store/store';
import { Page } from '../../router/types';
import { PATH } from '../../router/constants/paths';
import { deleteNotice } from '../../store/auth/slice';

import styles from './home.module.scss';

export default function Home(): ReactElement {
  const isNewUser = useSelector(selectIsNewUser);
  const dispatch = useAppDispatch();

  useEffect(() => {
    if (isNewUser) {
      const timer = setTimeout(() => {
        dispatch(deleteNotice());
      }, 3000);

      return () => {
        clearTimeout(timer);
      };
    }
  }, [isNewUser, dispatch]);

  return (
    <div className={styles.home} data-testid="home">
      <div className={styles.home__content}>
        <div className={classnames(styles.home__content_container, styles.links)}>
          <Link to={PATH[Page.Home]} className={classnames(styles.home__link, 'active')}>
            Main
          </Link>
          <Link to={PATH[Page.Catalog]} className={styles.home__link}>
            Catalog
          </Link>
          <Link to={PATH[Page.Login]} className={styles.home__link}>
            Login
          </Link>
          <Link to={PATH[Page.Register]} className={styles.home__link}>
            Register
          </Link>
          <Link to={PATH[Page.Profile]} className={styles.home__link}>
            Profile
          </Link>
          <Link to={PATH[Page.Cart]} className={styles.home__link}>
            Cart
          </Link>
          <Link to={PATH[Page.About]} className={styles.home__link}>
            About
          </Link>
        </div>
      </div>

      <div className={styles.home__content}>
        <div className={classnames(styles.home__content_container, styles.info)}>
          <p>
            To keep the project's concept clean, we use word ⭑<em>"Artists"</em>⋆｡★ instead of "Brands".
          </p>
          <p>
            For ☆｡<em>discounted</em>⭑ products check these artists: Cédrix Crespel, KAWS
          </p>
          <p>
            For ⭑<em>a single image</em>⋆｡★ in the slider check these artists: Jef Verheyen, Masoami Raku
          </p>
          <p>
            For <em>a promo code</em> use these codes: ART (-10%) and Special (-5%) for all items in a cart
          </p>
        </div>
      </div>

      {isNewUser && (
        <div className={styles.message}>
          <span>Hello and welcome!ヾ(☆▽☆)</span>
        </div>
      )}
    </div>
  );
}
