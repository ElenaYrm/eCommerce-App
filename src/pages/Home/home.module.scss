@import '../../styles/common/';

.home {
  padding: 10rem 2rem 8rem;
  width: 100%;
  height: 100%;
  background-image: url('/images/bg-home.jpg');
  background-size: cover;
  background-position: left;
  background-repeat: no-repeat;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1rem;

  &__content {
    max-width: 52rem;
    width: 100%;

    @media all and (max-width: 480px) {
      padding: 0 1rem;
    }

    &_container {
      width: 100%;
      padding: 4rem 3rem;
      background-color: $light;
      border-radius: 4rem;
      border: 1px solid $dark;
      text-align: center;

      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      gap: 0.5rem;
    }
  }

  &__link {
    @include link();
    width: fit-content;
    margin: 0 auto;
    padding: 1px;
    position: relative;
  }
}

.info {
  gap: 2rem;

  em {
    text-decoration: underline;
    font-weight: 500;
  }
}

.message {
  max-width: 37rem;
  width: 100%;
  padding: 0.8rem 2rem;
  background-color: $primary;
  border-radius: $border-radius-btn;
  box-shadow: 0 5px 5px rgba($dark, 0.2);
  color: $light;

  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  bottom: 6rem;

  transition: $transition-set;
}
