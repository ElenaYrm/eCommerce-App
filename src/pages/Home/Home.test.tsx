import { render, screen } from '@testing-library/react';
import { Home } from '../Home';
import * as reduxHooks from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

jest.mock('react-redux');
jest.mock('../../constant/metaData', () => {});
jest.mock('../../services/sdk/auth/methods', () => {});
jest.mock('../../services/sdk/product/methods', () => {});
jest.mock('../../services/sdk/catalog/methods', () => {});
jest.mock('../../services/sdk/customer/methods', () => {});

const useDispatchSpy = jest.spyOn(reduxHooks, 'useDispatch');

describe('Test Home page: ', () => {
  test('should render Home page without success register message', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(false);
    useDispatchSpy.mockReturnValue(jest.fn());
    const component = render(
      <MemoryRouter initialEntries={['/']}>
        <Home />
      </MemoryRouter>,
    );
    expect(screen.getByTestId('home')).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });

  test('should render Home page with success register message', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue(true);
    useDispatchSpy.mockReturnValue(jest.fn());
    const component = render(
      <MemoryRouter initialEntries={['/']}>
        <Home />
      </MemoryRouter>,
    );
    expect(screen.getByText('Hello and welcome!ヾ(☆▽☆)')).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });
});
