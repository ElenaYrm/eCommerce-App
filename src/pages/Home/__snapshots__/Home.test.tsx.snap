// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home page:  should render Home page with success register message 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
        data-testid="home"
      >
        <div
          class="home__content"
        >
          <a
            class="home__link active"
            href="/"
          >
            Main
          </a>
          <a
            class="home__link"
            href="/catalog"
          >
            Catalog
          </a>
          <a
            class="home__link"
            href="/login"
          >
            Login
          </a>
          <a
            class="home__link"
            href="/register"
          >
            Register
          </a>
          <a
            class="home__link"
            href="/profile"
          >
            Profile
          </a>
          <a
            class="home__link"
            href="/cart"
          >
            Cart
          </a>
        </div>
        <div
          class="home__content"
        >
          <div
            class="home__content_container"
          >
            <p>
              To keep the project's concept clean, we use word 
              <em>
                "Artists"
              </em>
               instead of "Brands".
            </p>
            <p>
              For 
              <em>
                discounted
              </em>
               products check these artists: Cédrix Crespel, KAWS
            </p>
            <p>
              For 
              <em>
                a single image
              </em>
               in the gallery check these artists: Jef Verheyen, Masoami Raku
            </p>
          </div>
        </div>
        <div
          class="message"
        >
          <span>
            Hello and welcome!ヾ(☆▽☆)
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
      data-testid="home"
    >
      <div
        class="home__content"
      >
        <a
          class="home__link active"
          href="/"
        >
          Main
        </a>
        <a
          class="home__link"
          href="/catalog"
        >
          Catalog
        </a>
        <a
          class="home__link"
          href="/login"
        >
          Login
        </a>
        <a
          class="home__link"
          href="/register"
        >
          Register
        </a>
        <a
          class="home__link"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="home__link"
          href="/cart"
        >
          Cart
        </a>
      </div>
      <div
        class="home__content"
      >
        <div
          class="home__content_container"
        >
          <p>
            To keep the project's concept clean, we use word 
            <em>
              "Artists"
            </em>
             instead of "Brands".
          </p>
          <p>
            For 
            <em>
              discounted
            </em>
             products check these artists: Cédrix Crespel, KAWS
          </p>
          <p>
            For 
            <em>
              a single image
            </em>
             in the gallery check these artists: Jef Verheyen, Masoami Raku
          </p>
        </div>
      </div>
      <div
        class="message"
      >
        <span>
          Hello and welcome!ヾ(☆▽☆)
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Home page:  should render Home page without success register message 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
        data-testid="home"
      >
        <div
          class="home__content"
        >
          <a
            class="home__link active"
            href="/"
          >
            Main
          </a>
          <a
            class="home__link"
            href="/catalog"
          >
            Catalog
          </a>
          <a
            class="home__link"
            href="/login"
          >
            Login
          </a>
          <a
            class="home__link"
            href="/register"
          >
            Register
          </a>
          <a
            class="home__link"
            href="/profile"
          >
            Profile
          </a>
          <a
            class="home__link"
            href="/cart"
          >
            Cart
          </a>
        </div>
        <div
          class="home__content"
        >
          <div
            class="home__content_container"
          >
            <p>
              To keep the project's concept clean, we use word 
              <em>
                "Artists"
              </em>
               instead of "Brands".
            </p>
            <p>
              For 
              <em>
                discounted
              </em>
               products check these artists: Cédrix Crespel, KAWS
            </p>
            <p>
              For 
              <em>
                a single image
              </em>
               in the gallery check these artists: Jef Verheyen, Masoami Raku
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
      data-testid="home"
    >
      <div
        class="home__content"
      >
        <a
          class="home__link active"
          href="/"
        >
          Main
        </a>
        <a
          class="home__link"
          href="/catalog"
        >
          Catalog
        </a>
        <a
          class="home__link"
          href="/login"
        >
          Login
        </a>
        <a
          class="home__link"
          href="/register"
        >
          Register
        </a>
        <a
          class="home__link"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="home__link"
          href="/cart"
        >
          Cart
        </a>
      </div>
      <div
        class="home__content"
      >
        <div
          class="home__content_container"
        >
          <p>
            To keep the project's concept clean, we use word 
            <em>
              "Artists"
            </em>
             instead of "Brands".
          </p>
          <p>
            For 
            <em>
              discounted
            </em>
             products check these artists: Cédrix Crespel, KAWS
          </p>
          <p>
            For 
            <em>
              a single image
            </em>
             in the gallery check these artists: Jef Verheyen, Masoami Raku
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
